presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -101,15 +102,34 @@ public void testShowCreateTable()
     @Test
     public void testDecimal()
     {
-        assertUpdate("CREATE TABLE test_decimal_short (x decimal(3,2))");
-        assertQueryFails("INSERT INTO test_decimal_short VALUES (decimal '3.14')", "Writing to columns of type decimal not yet supported");
-//        assertQuery("SELECT * FROM test_decimal_short", "SELECT CAST('3.14' AS DECIMAL(3,2))");
-        dropTable(getSession(), "test_decimal_short");
-
-        assertUpdate("CREATE TABLE test_decimal_long (x decimal(25,2))");
-        assertQueryFails("INSERT INTO test_decimal_long VALUES (decimal '3.14')", "Writing to columns of type decimal not yet supported");
-//        assertQuery("SELECT * FROM test_decimal_long", "SELECT CAST('3.14' AS DECIMAL(25,2))");
-        dropTable(getSession(), "test_decimal_long");
+        for (int precision = 1; precision <= 38; precision++) {
+            testDecimalWithPrecisionAndScale(precision, precision - 1);
+        }
+
+        for (int scale = 1; scale < 37; scale++) {
+            testDecimalWithPrecisionAndScale(38, scale);
+        }
+
+        for (int scale = 1; scale < 17; scale++) {
+            testDecimalWithPrecisionAndScale(18, scale);
+        }
+    }
+
+    private void testDecimalWithPrecisionAndScale(int precision, int scale)
+    {
+        checkArgument(precision >= 1 && precision <= 38, "Decimal precision must be between 1 and 38 inclusive, but was %d", precision);
```

Comment:
`Preconditions` only support `%s` for formatting.

Note that in general, `%s` is preferred, since it works for all types. Use the others like `%02d` when you need special type-specific formatting.

Reply:
We could shorten these messages by putting the values inline
```java
"Decimal precision (%s) must be between 1 and 38 inclusive"
"Decimal scale (%s) must be less than the precision (%s) and non-negative"

Reply:
Good idea; done.